GPT/beta-commands/gpt-function-calling.py

```diff
@@ -0,0 +1,101 @@
+from talon import Module, actions, settings
+from typing import Callable
+import requests, os, json
+from ..gpt import notify
+from .gpt_callables import function_specs, search_for_command, notify_user, display_response
+
+mod = Module()
+
+def gpt_function_query(prompt: str, content: str, insert_response: Callable[[str], str]) -> str:
+    notify("GPT Task Started")
+
+    try:
+        TOKEN = os.environ["OPENAI_API_KEY"]
+    except:
+        notify("GPT Failure: env var OPENAI_API_KEY is not set.")
+        raise Exception("GPT Failure")
+
+    url = "https://api.openai.com/v1/chat/completions"
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {TOKEN}",
+    }
+    data = {
+        "messages": [
+            {
+                "role": "system",
+                "content": f"language:\n{actions.code.language()}",
+            },
+            {"role": "user", "content": f"{prompt}:\n{content}"},
+        ],
+        "tools": function_specs,
+        "max_tokens": 2024,
+        "temperature": 0.6,
+        "n": 1,
+        "stop": None,
+        "model": settings.get("user.openai_model"),
+    }
+
+    response = requests.post(url, headers=headers, data=json.dumps(data))
+
+    if response.status_code == 200:
+        notify("GPT Task Completed")
+        message = response.json()["choices"][0]["message"]
+        process_function_calls(insert_response, message)
+
+        content = (message["content"] or "").strip()
+        if len(content) != 0:
+            display_response(content)
```

Comment:
We don't have to change this, but I personally feel like displaying the response in HTML isn't particularly useful unless its a help menu or something.

Reply:
I have found this useful because it gives me feedback on my prompt.  I don't like to see the response pasted into the text buffer, but if I didn't succeed it getting a function to be called then I do like to see the response still.

Reply:
Ok we will leave it. I was thinking the imgui used for the shell might be better, but that isn't accessible so I am fine with whatever.